File: Python_code_challenges/almostPalindrome.py

"""
The Palindromic score of a string is the number of errors (characters which
do not match) when the string is read forwards and backwards. For example, 
the palindromic score of 'fox' is 2, because 'fox' and 'xof' differ by two
characters. Write a function to take a string and return its palindromic 
score. 
"""

def apalin(word):
    palIndex = 0
    for i in range(len(word)):
        if word[i] != word[-(i+1)]:
            palIndex += 1
    return palIndex


print(apalin('abba')) #0
print(apalin('abcdcaa')) #2
print(apalin('fox')) #2
print(apalin('aaabbb')) #6

File: PHP/PHP_challenges/almostPalindrome.php
<?php

/*
The Palindromic score of a string is the number of errors (characters which
do not match) when the string is read forwards and backwards. For example, 
the palindromic score of 'fox' is 2, because 'fox' and 'xof' differ by two
characters. Write a function to take a string and return its palindromic 
score. 
*/

class Solution {
    function apalin($word) {
        $palindex = 0;
        $lenStr = strlen($word);
        foreach (range(0, $lenStr-1) as $i) {
            if ($word[$i] !== $word[($lenStr-1)-$i])
                $palindex += 1;
        }
        return $palindex;
    }
}


echo Solution::apalin('abba'), "\n"; #0
echo Solution::apalin('abcdcaa'), "\n"; #2
echo Solution::apalin('fox'), "\n"; #2
echo Solution::apalin('aaabbb'), "\n"; #6

